src/app/services/user-service/user-service.ts

```diff
@@ -23,13 +23,13 @@ export class UserService implements OnDestroy {
           payload: {
             account: {
               name,
-              username,
-              idTokenClaims: { oid }
+              idTokenClaims: { oid, preferred_username, email }
             },
             accessToken
           }
         } = result;
-        this.logIn({ name, username, oid, accessToken });
+        let userEmail = email || preferred_username;
+        this.logIn({ name, userEmail, oid, accessToken });
```

Comment:
Should we be allowing them to use the email to log in? Surely it should stay as user name?

Reply:
Its from the token after logging in, both of these fields are emails even user_name but username was populated by the upn which can change and shouldnt be used to identify a user. Where as Open ID allows the optional use of email which we have added and as a fail safe, replicate VTA in using the preferred_username